pulsar-client/src/main/java/org/apache/pulsar/client/impl/ZeroQueueConsumerImpl.java

```diff
@@ -215,4 +215,18 @@ protected void setCurrentReceiverQueueSize(int newSize) {
         //receiver queue size is fixed as 0.
         throw new NotImplementedException("Receiver queue size can't be changed in ZeroQueueConsumerImpl");
     }
+
+    @Override
+    protected void processPayloadByProcessor(BrokerEntryMetadata brokerEntryMetadata,
+                                             MessageMetadata messageMetadata, ByteBuf byteBuf,
+                                             MessageIdImpl messageId, Schema<T> schema,
+                                             int redeliveryCount, List<Long> ackSet, long consumerEpoch) {
+        if (this.isBatch(messageMetadata)) {
+            this.receiveIndividualMessagesFromBatch(brokerEntryMetadata,
+                    messageMetadata, redeliveryCount, ackSet, byteBuf, null, null, consumerEpoch, false);
```

Comment:
refactor the receiveIndividualMessagesFromBatch method and extract an error handling method instead of calling receiveIndividualMessagesFromBatch here. The call to receiveIndividualMessagesFromBatch was causing the confusion in the initial version of the code and this problem would be resolved by extracting the error handling method. It could be called `rejectBatchMessageByClosingConsumer` etc.